### PR TITLE
Added Helm chart

### LINKED ISSUE #8: Helm chart
## Description
The wallet must be deployable in a kubernetes setup using a helm chart for ease of deployment.

## Acceptance criteria
- Assume cloud native pg will be used
- Registries url must be configurable

## Scope limit
- The helm chart should contain the minimum required values to work
- Registries will only be hosted locally for now

## Solution plan
- [x] Add the chart
- [x] Add workflow to release chart
- [x] Add chart to artifacthub
- [ ] rewrite test https://github.com/project-origin/wallet/pull/17#discussion_r1200578159

### PR DESCRIPTION
**Pull request by MartinSchmidt**
Features added:
- Helm chart that can be installed in a k8s cluster for the wallet, fixes #8 
- Added --migrate flag to server to enable running migration
- Added --serve flag to server to enable starting server
- Database migrations are run in a separate job container to ensure they are only run once.
- Added workflow to ensure helm chart works.

### CODE REVIEW

File: charts/project-origin-wallet/templates/deployment-wallet.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: po-wallet-deployment
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: po-wallet
```
Comment (duizer) [inline charts/project-origin-wallet/templates/deployment-wallet.yaml:8]: Should this label be on all k8s objects in this chart?

File: charts/project-origin-wallet/templates/deployment-wallet.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+spec:
+  replicas: 1
```
Comment (duizer) [inline charts/project-origin-wallet/templates/deployment-wallet.yaml:10]: Why only 1? 

```suggestion
  replicas: 3
```

File: charts/project-origin-wallet/templates/deployment-wallet.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                  name: {{ .Values.persistance.cloudNativePG.name }}-app
+                  key: username
+            - name: DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.persistance.cloudNativePG.name }}-app
+                  key: password
+            - name: ConnectionStrings__Database
+              value: Host=$(DB_HOST); Port=$(DB_PORT); Database=$(DB_DATABASE); Username=$(DB_USERNAME); Password=$(DB_PASSWORD);
+          {{ end }}
```
Comment (duizer) [inline charts/project-origin-wallet/templates/deployment-wallet.yaml:53]: Should there be an else-case for when not using cloudNativePG?

File: charts/project-origin-wallet/templates/migrate-job.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  backoffLimit: 0
+  parallelism: 1
+  completions: 1
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: po-wallet-migrate-job
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        args: ["--migrate"]
```
Comment (duizer) [inline charts/project-origin-wallet/templates/migrate-job.yaml:17]: args is written differently here than in the deployment

File: charts/project-origin-wallet/templates/deployment-wallet.yaml (hunk 1 above)
Comment (duizer) [reply charts/project-origin-wallet/templates/deployment-wallet.yaml:8]: Nevermind

File: charts/project-origin-wallet/values.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# Default values for project-origin-wallet.
+
+# image defines the image configuration for the wallet server
+image:
+
+  # repository defines the image repository to use
+  repository: ghcr.io/project-origin/wallet-server
+
+  # tag defines the image tag to use, defaults to the chart appVersion
+  tag:
```
Comment (duizer) [inline charts/project-origin-wallet/values.yaml:10]: Is it possible to set a default value for the tag?

File: .github/workflows/pullrequest-verify-chart.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  verify-chart-with-latest-container:
+    name: Verify chart with local built container
+    runs-on: ubuntu-latest
```
Comment (duizer) [inline .github/workflows/pullrequest-verify-chart.yaml:11]: Should version be pinned? There is a (small) chance that future versions will not support kind, helm and docker...

File: charts/project-origin-wallet/values.yaml (hunk 5 above)
Comment (MartinSchmidt) [reply charts/project-origin-wallet/values.yaml:10]: it is set through the AppVersion in the chart, this is only to document that override is allowed.

File: charts/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,24 @@
+FROM debian:bullseye-20230522 as downloader
```
Comment (duizer) [inline charts/Dockerfile:1]: What is this Dockerfile used for?

Comment (MartinSchmidt) [reply charts/Dockerfile]: I removed it again, it was just a commit errror

**Review by duizer — APPROVED** [review]

**Pull request merged**